.travis.yml

```diff
@@ -1,18 +1,29 @@
-sudo: false
-
 language: rust
-
-rust:
-  - stable
-  - nightly
+cache: cargo
 
 matrix:
-  allow_failures:
-    - rust: nightly
-
-before_script:
-  - rustup component add clippy rustfmt
+  include:
+  - rust: nightly
+    env: NAME=test-after-dark
+    script:
+    - cargo test --all --verbose
+  - rust: 1.36.0
+    env: NAME=test
+    script:
+    - cargo test --all --verbose
+  - rust: 1.36.0
+    env: NAME=pretty
+    before_script:
+    - rustup component add rustfmt
+    script:
+    - cargo fmt --all -- --check
+  - rust: 1.36.0
+    env: NAME=petty
+    before_script:
+    - rustup component add clippy
+    script:
+    - cargo clippy --all --all-targets --verbose
```

Comment:
Why drop `--deny warnings`?

Reply:
Because from my experience it's very annoying for new contributors to have their build fail for something like ["weird underscore placement"](https://rust-lang.github.io/rust-clippy/v0.0.212/index.html#inconsistent_digit_grouping) (and similar).

I'd rather suggest you add explicit `#![deny(clippy::thing_i_never_want_to_allow)]` lines, which makes this fail for the combined set of lints the clippy maintainers consider bad enough to set to deny and the ones you consider bad enough to deny. You can always check the logs for warnings later and fix them up in a cleanup PR.

Reply:
But this check is primarily run before merging, locally people can use `cargo check` or `cargo clippy` without denying the warnings. Hence, I'd prefer avoiding the separate clean-up MR and keeping things clean from the outset.

Reply:
I'm not gonna tell you what to do but I want to point out two things:

1. You seem to want to have more failed builds. Seeing a failed build is very discouraging to new contributors. When people are doing a good job and made a nice PR, it's easy to point them at the warnings in the CI log. This will not feel like "shit, I broke it, nobody will look at this PR" but like "oh, there's an additional set of feedback how to improve my code".
2. The clippy authors enable the lints they consider to be severe to deny by default, the rest are suggestions. Opting in to all suggestions is probably not wise. I can only repeat my suggestion to set this in `lib.rs`, which will be less surprising to developers and gives you the chance to also comment on why you want to deny a specific lint.

Reply:
I'll merge this and see how it fares. Thanks!